test/error/error.test.js

```diff
@@ -16,85 +17,130 @@ var xml = '<scxml xmlns="http://www.w3.org/2005/07/scxml" version="1.0"\n\
+		var error = {}
+		var parser = new DOMParser({
+			errorHandler: function (key, msg) {
+				error[key] = msg
+			},
+		})
+		var doc = parser.parseFromString(
+			'<html disabled><1 1="2"/></body></html>',
+			'text/xml'
+		)
+		assert.isTrue(
+			typeof error.warning === 'string',
+			'error.warning: ' + String(error.warning)
+		)
+		assert.isTrue(
+			typeof error.error === 'string',
+			'error.error: ' + String(error.error)
+		)
+	})
 
-  it("only one function with one argument builds list", () => {
-  	var error = []
-    var parser = new DOMParser({
-    	errorHandler:function(msg){error.push(msg)}
-	});
-    	var doc = parser.parseFromString('<html disabled><1 1="2"/></body></html>', 'text/xml');
-    	error.map(function(e){error[/^\[xmldom (\w+)\]/.exec(e)[1]]=e})
-		assert.isTrue(typeof error.warning === 'string', 'error.warning:'+error.warning);
-		assert.isTrue(typeof error.error === 'string', 'error.error:'+error.error);
-  })
+	it('only one function with one argument builds list', () => {
+		var error = []
+		var parser = new DOMParser({
+			errorHandler: function (msg) {
+				error.push(msg)
+			},
+		})
+		var doc = parser.parseFromString(
+			'<html disabled><1 1="2"/></body></html>',
+			'text/xml'
+		)
+		error.map(function (e) {
+			error[/^\[xmldom (\w+)\]/.exec(e)[1]] = e
+		})
+		assert.isTrue(
+			typeof error.warning === 'string',
+			'error.warning:' + error.warning
+		)
+		assert.isTrue(typeof error.error === 'string', 'error.error:' + error.error)
+	})
 
-  it("compare one function with only one key", () => {
-  	var error = []
-  	var errorMap = []
-		const faulty = '<html><body title="1<2">test</body></html>';
+	it('compare one function with only one key', () => {
+		var error = []
+		var errorMap = []
+		const faulty = '<html><body title="1<2">test</body></html>'
 		new DOMParser({
-    	errorHandler:function(msg){error.push(msg)}
-	}).parseFromString(faulty, 'text/xml');
-    ['warn','warning','error','fatalError'].forEach(function(k){
-    	var errorHandler = {[k]:[]};
-    	errorHandler[k] = function(msg){errorMap[k].push(msg)}
-	    new DOMParser({errorHandler:errorHandler}).parseFromString(faulty, 'text/xml');
-    	assert.isTrue(errorHandler[k].length > 0, 'expected entries for '+k);
-    });
-    var error2 = [];
-    for(var n in errorMap){
-    	error2 = error2.concat(errorMap[n])
-    	assert(error.length, errorMap[n].length)
-    }
-   
-    assert(error2.sort().join(','), error.sort().join(','), 'expected same messages')
-  })
-
-  it("error function throwing is not caught", () => {
-  	var error = []
-    var parser = new DOMParser({
-    	locator:{},
-    	errorHandler:{
-			error:function(msg){
-				error.push(msg);
-				throw new Error('from throwing errroHandler.error')
+			errorHandler: function (msg) {
+				error.push(msg)
+			},
+		}).parseFromString(faulty, 'text/xml')
+		Array.from(['warn', 'warning', 'error', 'fatalError']).forEach((k) => {
```

Comment:
FYI I used `Array.from` here, as a trick to avoid the automatic semicolon insertion (ASI) by Prettier.

We should probably add some blank lines to this test case as well.